Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/InteractableToggleCollection.cs

```diff
@@ -11,78 +12,108 @@ namespace Microsoft.MixedReality.Toolkit.UI
+        [Tooltip("Currently selected index in the ToggleList, default is 0")]
+        [SerializeField]
+        private int currentIndex;
 
-        private void Start()
+        /// <summary>
+        /// The current index in the array of interactable toggles
+        /// </summary>
+        public int CurrentIndex
         {
-            for (int i = 0; i < ToggleList.Length; ++i)
+            get
             {
-                int itemIndex = i;
-                // add selection event handler to each button
-                ToggleList[i].OnClick.AddListener(() => OnSelection(itemIndex));
-                ToggleList[i].CanDeselect = false;
+                return currentIndex;
+            }
+            set
+            {
+                SetSelection(value, false, true);
             }
-
-            OnSelection(CurrentIndex, true);
         }
 
-        private void OnEnable()
+        [Tooltip("This event is triggered when any of the toggles in the ToggleList are selected")]
+        /// <summary>
+        /// This event is triggered when any of the toggles in the ToggleList are selected
+        /// </summary>
+        public UnityEvent OnSelectionEvents = new UnityEvent();
+
+        private void Start()
         {
-            OnSelection(CurrentIndex, true);
+            if (ToggleList != null)
+            {
+                // Add listeners to each toggle in ToggleList
+                for (int i = 0; i < ToggleList.Length; ++i)
+                {
+                    int itemIndex = i;
+                    ToggleList[i].OnClick.AddListener(() => SetSelection(itemIndex, true, false));
+                    ToggleList[i].CanDeselect = false;
+                }
+
+                SetSelection(CurrentIndex, true, true);
+            }
         }
 
         /// <summary>
-        /// Sets the selected index and selected Interactive
+        /// Set the selection of a an element in the toggle collection based on index.
+        /// <param name="index">Index of an element in ToggleList</param>
+        /// <param name="force">Force update currentIndex</param>
+        /// <param name="fireOnClick">The manual trigger of the OnClick event. OnClick event is manually triggered 
+        /// when the CurrentIndex is updated via script or inspector</param>
         /// </summary>
-        public void SetSelection(int index)
+        public void SetSelection(int index, bool force = false, bool fireOnClick = false)
         {
-            if (!isActiveAndEnabled ||
-                (index < 0 || ToggleList.Length <= index))
+            if (index < 0 || ToggleList.Length <= index)
             {
+                Debug.LogWarning("Index out of range of ToggleList: " + index);
                 return;
             }
 
-            OnSelection(index, true);
+            if (CurrentIndex != index || force)
+            {
+                currentIndex = index;
+
+                OnSelection(index);
+            }
+
+            if (fireOnClick)
```

Comment:
I believe this needs to be inside the if statement body above. If we aren't changing selection, we shouldn't be firing a click? @julenka thoughts?

Reply:
Like if I call CurrentIndex =2, many times over, seems weird to call onclick?

Reply:
I think Catherine had a specific case she was fixing which required onclick to get called, I think otherwise the visuals didn't update? I don't fully remember...

Reply:
There was no specific case.  The fireOnClick was moved into the index changed and force check.  After the move, if I changed the index from the inspector at runtime the toggle collection was not updated.  Setting force to true in the Inspector script solved this issue.